packages/@glimmer/runtime/lib/render.ts

```diff
@@ -27,3 +31,69 @@ export default function render<T>(
   let vm = VM.initial(program, env, self, dynamicScope, builder, handle);
   return new TemplateIteratorImpl(vm);
 }
+
+export type RenderComponentArgs = Dict<PathReference<Opaque>>;
+
+/**
+ * Returns a TemplateIterator configured to render a root component.
+ */
+export function renderComponent<T>(
+  program: RuntimeProgram<T>,
+  env: Environment,
+  builder: ElementBuilder,
+  main: number,
+  name: string,
+  args: RenderComponentArgs = {}
+): TemplateIterator {
+  const vm = VM.empty(program, env, builder, main);
+  const { resolver } = vm.constants;
+
+  const definition = expect(
+    resolveComponent(resolver, name, null),
+    `could not find component "${name}"`
+  );
+
+  const { manager, state } = definition;
+  const capabilities = capabilityFlagsFrom(manager.getCapabilities(state));
+
+  let invocation;
+
+  if (hasStaticLayoutCapability(capabilities, manager)) {
+    invocation = manager.getLayout(state, resolver);
+  } else {
+    throw new Error('Cannot invoke components with dynamic layouts as a root component.');
+  }
+
+  // Get a list of tuples of argument names and references, like
+  // [['title', reference], ['name', reference]]
+  const argList = Object.keys(args).map(key => [key, args[key]]);
+
+  const blockNames = ['main', 'else', 'attrs'];
+  // Prefix argument names with `@` symbol
+  const argNames = argList.map(([name]) => `@${name}`);
+
+  vm.pushFrame();
```

Comment:
I'm trying to remember if there is a `popFrame` when the iterator is exhausted or not. Do you happen to know?

Reply:
I walked through the opcodes the final set of ops the vm is running in order are,

PopFrame, PopScope, PopDynamicScope, CommitComponentTransaction, Return